build-tools/packages/version-tools/src/test/internalVersionScheme.test.ts

```diff
@@ -387,4 +388,40 @@ describe("internalScheme", () => {
 			);
 		});
 	});
+
+	describe("checking test version schema", () => {
+		it("0.0.0-285010-test is test version", () => {
+			const input = `0.0.0-285010-test`;
+			const result = isInternalTestVersion(input);
+			assert.isTrue(result);
+		});
+
+		it("0.0.0-test-285010 is test version", () => {
+			const input = `0.0.0-test-285010`;
+			const result = isInternalTestVersion(input);
+			assert.isFalse(result);
+		});
+
+		it("2.1.0-test-285010 is not test version", () => {
+			const input = `2.1.0-test-285010`;
+			const result = isInternalTestVersion(input);
+			assert.isFalse(result);
+		});
+
+		it("2.1.0-285010-test is not test version", () => {
+			const input = `2.1.0-285010-test`;
+			const result = isInternalTestVersion(input);
+			assert.isFalse(result);
+		});
+
+		it("2.1.0-285010 is not test version", () => {
+			const input = `2.1.0-285010`;
+			assert.throws(() => isInternalTestVersion(input));
+		});
+
+		it("2.1.0 is not test version", () => {
+			const input = `2.1.0`;
+			assert.throws(() => isInternalTestVersion(input));
+		});
```

Comment:
I would not describe these cases as not test versions. They don't return false.
Apart from the numbers I think it is helpful for the test cases to use words describing what the pattern represents. `2.1.0` is a "release" version. I would say all of the others are "prerelease" versions and some(?) of those are "internal".